### PR TITLE
[sw/silicon_creator] Harden remaining lc_state switches

### PR DESCRIPTION
**Pull request by alphan**
Signed-off-by: Alphan Ulusoy <alphan@google.com>

### CODE REVIEW

**Comment by alphan** [comment]
[disassembly](https://gist.github.com/8d37c4e28ee0872f3ccae4bb1fd88ba5)

File: sw/device/silicon_creator/lib/sigverify.c [hunk 1]
```diff
@@ -201,26 +201,29 @@ static rom_error_t sigverify_encoded_message_check(
...
  * @param lc_state Life cycle state of the device.
- * @param[out] use_sw Use software implementation for signature verification.
- * @return Result of the operation.
+ * @return Whether to use software implementation for signature verification.
  */
-static rom_error_t sigverify_use_sw_rsa_verify(lifecycle_state_t lc_state,
-                                               hardened_bool_t *use_sw) {
-  switch (lc_state) {
+static hardened_bool_t sigverify_use_sw_rsa_verify(lifecycle_state_t lc_state) {
+  switch (launder32(lc_state)) {
```
Comment (moidx) [inline sw/device/silicon_creator/lib/sigverify.c:207]: How about doing this instead? Assuming is difficult for the attacker to trigger `kLcStateTest`.

```cc
if (launder32(lc_state) == kLcStateTest) {
  HARDENED_CHECK_EQ(lc_state, kLcStateTest);
  return kHardenedBoolTrue;
}
return otp_read32(OTP_CTRL_PARAM_CREATOR_SW_CFG_USE_SW_RSA_VERIFY_OFFSET);
```

**Review by moidx — APPROVED** [review]

File: sw/device/silicon_creator/lib/sigverify.c (hunk 1 above)
Comment (alphan) [reply sw/device/silicon_creator/lib/sigverify.c:207]: Discussed this offline, will keep the switch statement as a redundant check for the life cycle state.

File: sw/device/silicon_creator/lib/sigverify_unittest.cc [hunk 2]
```diff
@@ -219,15 +219,16 @@ TEST_F(SigverifyInTestStates, BadSignatureIbex) {
...
 
-TEST_F(SigverifyInInvalidStates, BadLcState) {
-  uint32_t flash_exec = 0;
-  EXPECT_EQ(
-      sigverify_rsa_verify(&kSignature, &key_, &kTestDigest,
-                           static_cast<lifecycle_state_t>(0), &flash_exec),
-      kErrorSigverifyBadLcState);
-  EXPECT_EQ(flash_exec, std::numeric_limits<uint32_t>::max());
+TEST_F(SigverifyBadLcStateDeathTest, BadLcState) {
+  ASSERT_DEATH(
```
Comment (mcy) [inline sw/device/silicon_creator/lib/sigverify_unittest.cc:225]: Nit: this can be EXPECT.

**Review by mcy — APPROVED** [review]

File: sw/device/silicon_creator/lib/sigverify_unittest.cc (hunk 2 above)
Comment (alphan) [reply sw/device/silicon_creator/lib/sigverify_unittest.cc:225]: #10549

**Pull request merged**